### ISSUE TITLE
Improve Invoice Upload experience

### ISSUE THREAD

**Issue by jonpugh** [HMSADMSYS/prompt#163]
From HMS:

> Upload invoice slow – users are experiencing delays and timeout issues when uploading invoice files.  Can you review the code and revise to improve upload speed?

**Tech approach:** We can't make the internet faster, but we can improve the experience.  Let's try to implement the progress meter and ensure that PHP execution time is set so users don't get timeouts.

**Notes:** 
- Code validation is probably the thing taking the most time.
- Currently, on submission of the upload form, the code loops through all the rows in the spreadsheet generating all the invoices at once, in a single form post.
  
  This is so that the process can be aborted if there is a problem with any single invoice.
  
  If we add a drupal batch process, we must be sure to do the same thing.  Each batch item should create an invoice, but at the end, if anything went wrong with any of the invoices, let's delete all of the nodes that were created.

**Comment by jonpugh** [comment]
This one is still outstanding. @varitree is going to take a look.

**Comment by jonpugh** [comment]
I think the batch uploader was too much at the moment. It was very challenging to get it to save data correctly, and to have the tests run properly.  

I've removed the batch uploader and replaced it with a module called "hide_submit". It offers a visual experience that locks the submit form and adds an animation. I think this might be enough for end users that are uploading big spreadsheets. I also made sure the timeout is set to unlimited so a big spreadsheet upload shouldn't time out.

Here's a screenshot of the new uploader:

![screenshot from 2016-06-23 14 26 30](https://cloud.githubusercontent.com/assets/106420/16316107/ddf43912-3952-11e6-85cf-547994286ab1.png)
